src/pools.jl

```diff
@@ -51,6 +51,35 @@ safesizehint!(x, n) = sizehint!(x, min(4096, n))
 # determines whether we'll look up object caches in .keyedvalues or .values
 iskeyed(::Pool{K}) where {K} = K !== Nothing
 
+Base.keytype(::Type{<:Pool{K}}) where {K} = K
+Base.keytype(p::Pool) = keytype(typeof(p))
+
+Base.valtype(::Type{<:Pool{<:Any, T}}) where {T} = T
+Base.valtype(p::Pool) = valtype(typeof(p))
```

Comment:
What are these for? I don't see that explained in the PR description or the tests.

Reply:
Makes sense. I think if you want to add these, you should just add some documentation explaining what these are and what they're for. Just needs some docstrings i think. Or split it into another PR 🤷